cfme/fixtures/cli.py

```diff
@@ -118,3 +129,331 @@ def app_creds_modscope():
         'sshlogin': credentials['ssh']['username'],
         'sshpass': credentials['ssh']['password']
     }
+
+
+@pytest.fixture
+def appliance_preupdate(old_version, appliance):
+    """Requests appliance from sprout based on old_versions, edits partitions and adds
+    repo file for update"""
+    series = appliance.version.series()
```

Comment:
Looks like we have duplication of line 138-153 in 3 fixtures. We can move it out to separate fixture like:
```@pytest.fixture
def get_app(appliance, old_version):
    def _get_app(app_number):
        series = appliance.version.series()
        update_url = "update_url_{}".format(series.replace('.', ''))
        usable = []
        sp = SproutClient.from_config()
        available_versions = set(sp.call_method('available_cfme_versions'))
        for a in available_versions:
            if a.startswith(old_version):
                usable.append(Version(a))
        usable_sorted = sorted(usable, key=lambda o: o.version)
        try:
            apps, pool_id = sp.provision_appliances(count=app_number, preconfigured=True,
                                                    provider_type='rhevm',
                                                    lease_time=180, version=str(usable_sorted[-1]))
        except Exception as e:
            logger.exception("Couldn't provision appliance with following error:{}".format(e))
            raise SproutException('No provision available')
        return apps, pool_id
    return _get_app